src/elements.js

```diff
@@ -488,8 +488,14 @@ _.register({
 			"time": /^[H\d]{2}:[M\d]{2}/i,
 			"datetime-local": /^[Y\d]{4}-[M\d]{2}-[D\d]{2} [H\d]{2}:[Mi\d]{2}/i,
 			"date": /^[Y\d]{4}-[M\d]{2}-[D\d]{2}$/i,
+			"duration": /^P/,
+        	"duration-normal": /(day[s]?|hour[s]?|minute[s]?|second[s]?)$/,
```

Comment:
Ah, I see what you meant now. Yes, we'd need to add ms, weeks, months, years too. 
One direction to explore: can we make this work for other languages beyond English? (look at the `Intl` object)